### DISCUSSION TITLE
How to set headers to json route on SSG

### DISCUSSION THREAD

**Discussion by slash9494** [vercel/vercel#8541] [Help]
I build pages with `getStaticProps` and get `.html`,`.json` files on server folder of `.next` 
and I set headers in `next.config.js` 
![스크린샷 2022-09-10 오전 12 25 04](https://user-images.githubusercontent.com/70849655/189386046-e9313196-862f-4910-8142-89738e12c327.png)

everything is ok  except for `[page].json` files
![스크린샷 2022-09-10 오전 12 32 02](https://user-images.githubusercontent.com/70849655/189387791-27287248-0c7e-49cd-90f3-955b68c4c536.png)
![스크린샷 2022-09-10 오전 12 32 17](https://user-images.githubusercontent.com/70849655/189387881-6aa1b9e8-97df-478d-9c9d-ef1be3bba84b.png)

Does anyone know how to set headers on JSON files ??

**Comment by amyegan** [comment]
Hi @slash9494! It looks like you're using a Serverless Function to generate JSON for your pages. [Cache Control](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching#cache-control) headers cannot be set using `next.config.js` or `vercel.json` for Serverless Functions. Instead, you should set the headers directly in the function, like
```
// pages/api/user.js

export default function handler(req, res) {
  res.setHeader('Cache-Control', 's-maxage=86400');
  res.status(200).json({ name: 'John Doe' });
}
```

**Reply by slash9494** [reply]
I got it, and I have some similar questions
if the latest other caches are on the edge cache, is my cache removed from the edge cache??
when I connected my service after 24 hours, x-vercel-cache was  "MISS" even though it had no new deployment

**Reply by amyegan** [reply]
I'm not sure I'm clear on what you're asking, but I'll answer based on my understanding of the question. Your first visit would be a [`MISS`](https://vercel.com/docs/concepts/edge-network/caching#miss) because nothing is available in cache yet. Subsequent visits should follow the `Cache-Control` settings unless a [new deployment invalidates the cache](https://vercel.com/docs/concepts/edge-network/caching#cache-invalidation).

You can find more information in [Caching](https://vercel.com/docs/concepts/edge-network/caching#) and [Caching responses from Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching#) documentation.
